### PR TITLE
208 building gains and loss diagram

### PR DESCRIPTION
**Pull request by daren-thomas**
This PR implements #208 (Building Gains and Loss Diagram). Test it with the gh file `hive\GrasshopperExamples\Hive_Results_Visualizer_Test.gh`. You should see something like this:

image

### CODE REVIEW

**Comment by daren-thomas** [comment]
@amraladdin you can use these files for testing:

[Libraries.zip](https://github.com/architecture-building-systems/hive/files/5022343/Libraries.zip)
[Hive_Results_Visualizer_Test.zip](https://github.com/architecture-building-systems/hive/files/5022347/Hive_Results_Visualizer_Test.zip)

**Comment by daren-thomas** [comment]
@amraladdin @christophwaibel I suggest, in case of omissions to create separate issues and merge instead of waiting here too long - my next task will probably build up on this one...

File: src/Hive.IO/TestVisualizer/Form1.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
+﻿using System;
```
Comment (christophwaibel) [inline src/Hive.IO/TestVisualizer/Form1.cs:1]: I don't fully get it... are you using this project here to load in `EnergyBalancePlots.cs` from `Hive.IO`?

File: src/Hive.IO/Hive.IO.sln [hunk 2]
```diff
@@ -5,6 +5,8 @@ VisualStudioVersion = 16.0.29326.143
 MinimumVisualStudioVersion = 10.0.40219.1
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Hive.IO", "Hive.IO\Hive.IO.csproj", "{1747808E-543C-4DA4-8FB1-B5EBC8B80C77}"
 EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "TestVisualizer", "TestVisualizer\TestVisualizer.csproj", "{CE013650-1F89-43FD-BB23-6DD17C1B63B7}"
```
[the remark this answers is not in the document]
Comment (daren-thomas) [reply src/Hive.IO/Hive.IO.sln:8]: yes, let's keep this project around while i finish up the remaining plots - it let's me code much faster as i don't have to start rhino, start grasshopper, click, click, click - _each_ time i want to test something. i'll probably also use something similar for coding the forms. we can delete them later.

File: src/Hive.IO/Hive.IO/Environment/Plots/DrawingExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,138 @@
+﻿using System;
```
[the remark this answers is not in the document]
Comment (daren-thomas) [reply src/Hive.IO/Hive.IO/Environment/Plots/DrawingExtensions.cs:1]: oops. i'll check that out.

File: .gitignore [hunk 4]
```diff
@@ -118,3 +118,21 @@ Python data_from_sensors/plot_data.py
 Mothercell/Hive.IO/.vs/Hive.IO/DesignTimeBuild/.dtbcache
 *.bak
 *.orig
+
```
[the remark this answers is not in the document]
Comment (daren-thomas) [reply .gitignore:121]: strange. i actually did a lot of work on .gitignore to get these out. i'll check it again. maybe they need to be deleted manually as it seems your git is tracking them. maybe mine was too.

File: src/Hive.IO/Hive.IO/DataHandling/ResultsPlotting.cs [hunk 5]
```diff
@@ -159,5 +160,21 @@ public double[] OperationEnergySystemsMonthly(bool normalized)
...
+        public float SolarGains => 100.0f;
+        public float InternalGains => 200f;
+        public float PrimaryEnergy => 300.0f;
+        public float RenewableEnergy => 400.0f;
+        public float Electricity => 110.0f;
+        public float VentilationLosses => 120.0f;
+        public float EnvelopeLosses => 50.0f;
+        public float WindowsLosses => 200f;
+        public float SystemLosses => 300f;
+        public float PrimaryTransferLosses => 123.0f;
```
[the remark this answers is not in the document]
Comment (daren-thomas) [reply src/Hive.IO/Hive.IO/DataHandling/ResultsPlotting.cs:175]: i have no idea. i'm just following the specs for the visualizer. @amraladdin is the building scientist. maybe the two of you can discuss this together?

File: src/Hive.IO/Hive.IO/Environment/Plots/EnergyBalancePlot.cs [hunk 6]
```diff
@@ -0,0 +1,76 @@
+﻿using System;
```
[the remark this answers is not in the document]
Comment (daren-thomas) [reply src/Hive.IO/Hive.IO/Environment/Plots/EnergyBalancePlot.cs:1]: sure. i'll fix that!

File: src/Hive.IO/Hive.IO/Plots/RectangleFExtensions.cs [hunk 7]
```diff
@@ -1,12 +1,14 @@
-﻿using System.Collections;
+﻿using System;
+using System.Collections;
 using System.Collections.Generic;
 using System.Drawing;
 using System.Linq;
 using Grasshopper.Kernel;
+using Grasshopper.Kernel.Types.Transforms;
```
[the remark this answers is not in the document]
Comment (daren-thomas) [reply src/Hive.IO/Hive.IO/Plots/RectangleFExtensions.cs:7]: i don't think so. in fact, i'm not sure i need any grasshopper stuff in that file. i'll take a look. wait... i thought that file was removed? (it should be `DrawingExtensions.cs` - I'll double-check and delete it if possible. good job with pruning the source tree!

File: src/Hive.IO/TestVisualizer/Form1.cs (hunk 1 above)
Comment (daren-thomas) [reply src/Hive.IO/TestVisualizer/Form1.cs:1]: yes. the point is to have the feedback loop (compile, run, test) as quick as possible to stay in the zone.

Comment (daren-thomas) [reply src/Hive.IO/TestVisualizer/Form1.cs:1]: (debugging is also easier)

File: .gitignore (hunk 4 above)
[the remark this answers is not in the document]
Comment (daren-thomas) [reply .gitignore:121]: ok. updated .gitignore, updated the git repository to remove those files (they were already added and `.gitignore` is not used for files already in the repository)

File: src/Hive.IO/Hive.IO/Environment/Plots/EnergyBalancePlot.cs (hunk 6 above)
[the remark this answers is not in the document]
Comment (daren-thomas) [reply src/Hive.IO/Hive.IO/Environment/Plots/EnergyBalancePlot.cs]: how did they end up there? they're not even the proper files. i removed them.

File: src/Hive.IO/Hive.IO/Plots/RectangleFExtensions.cs (hunk 7 above)
[the remark this answers is not in the document]
Comment (daren-thomas) [reply src/Hive.IO/Hive.IO/Plots/RectangleFExtensions.cs]: another weird one - this file was not showing up in the ide - i had renamed it with resharper. maybe something went wrong and it got committed anyway. i removed those using directives from `DrawingExtensions.cs`.

**Pull request merged**